packages/iTwinUI-react/src/core/Slider/Slider.test.tsx

```diff
@@ -532,6 +1131,53 @@ it('should move thumb when pointer down on rail', () => {
   expect(handleOnUpdate).toHaveBeenCalledWith([30]);
 });
 
+// TODO: I don't know what this test does and how to transition it for the vertical slider version
+// TODO: I added some unnecessary check conditions. Remove them
+// TODO: Don't know how to fix this test
+it('should move thumb when pointer down on rail (vertical)', () => {
+  const handleOnChange = jest.fn();
+  const handleOnUpdate = jest.fn();
+
+  const { container } = render(
+    verticalSliderWrapper(
+      <Slider
+        values={defaultSingleValue}
+        tickLabels={<span className='custom-tick-mark'>Custom</span>}
+        onChange={handleOnChange}
+        onUpdate={handleOnUpdate}
+        orientation='vertical'
+      />,
+    ),
+  );
+
+  assertBaseElement(container);
+  const sliderContainer = container.querySelector(
+    '.iui-slider-container',
+  ) as HTMLDivElement;
+  console.log(12345, sliderContainer.getBoundingClientRect());
+  expect(sliderContainer.getBoundingClientRect().left).toBe(10);
+  expect(sliderContainer.getBoundingClientRect().right).toBe(1010);
+  expect(sliderContainer.getBoundingClientRect().top).toBe(0);
+  expect(sliderContainer.getBoundingClientRect().bottom).toBe(60);
+  expect(sliderContainer.getBoundingClientRect().width).toBe(1000);
+  expect(sliderContainer.getBoundingClientRect().height).toBe(60);
+
+  /* fire a pointer down event 30% down the slider */
+  act(() => {
+    fireEvent.pointerDown(sliderContainer, {
+      pointerId: 5,
+      buttons: 1,
+      clientY: 30,
+    });
+  });
+
+  const thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb.getAttribute('aria-valuenow')).toEqual('30');
+
+  expect(handleOnChange).toHaveBeenCalledWith([30]);
+  expect(handleOnUpdate).toHaveBeenCalledWith([30]);
+});
```

Comment:
I was having trouble understanding what this test actually does and why is it failing for the vertical slider case. The below four other vertical slider test cases that have to do something with `clientX` or `clientY` too are also failing because of the same reason. I am trying to debug it in more detail, but is there any mistake that stands out? Do you have any advice?

Reply:
Have you found the answer because I see tests are passing?

Reply:
**[DONE]** Yes, I got them to pass! After you told where the size is being mocked, I understood why the five failing test were failing. It was failing because of the dimension checks mismatch and the mock clicks not being in the correct screen location and hence causing wrong values in the slider. I correct the expected dimensions, put the clicks in the correct `clientY` value that caused the actual slider values match the expected slider value after the click.